components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -580,6 +614,266 @@ protected void initiateAuthenticationRequest(HttpServletRequest request, HttpSer
+                        " found. Query parameter: " + OIDCAuthenticatorConstants.FEDERATED_TOKEN_SCOPE +
+                        " value: " + requestedScopesViaQueryParams + " found for federated token sharing, IDP: "
+                        + authenticatorName);
+            }
+            // Checks the scopes requested via authorize request query parameters.
+            return requestedScopesViaQueryParams;
+        }
+    }
+
+    /**
+     * This method returns the adaptive script federated authenticator param value for a given parameter name.
+     *
+     * @param context                The authentication context with federated authenticator params.
+     * @param authenticatorParamName The federated authenticator parameter name.
+     * @return The adaptive script federated authenticator param value for the given parameter name.
+     */
+    private String getAdaptiveScriptValues(AuthenticationContext context, String authenticatorParamName) {
+
+        Map<String, String> runtimeParams = this.getRuntimeParams(context);
+        if (runtimeParams != null) {
+            return runtimeParams.get(authenticatorParamName);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    /**
+     * The optional scope string cannot have scattered segments for the same authenticator.
+     * Only the very first segment is considered.
+     * i.e. A valid string:
+     * Google Calender has read write scopes, Microsoft Authenticator has https://googleapis.calender scope
+     * A valid string:
+     * federated_token_scope=Google Calander;read write,Microsoft Authenticator;https://googleapis.calender
+     * A valid string:
+     * federated_token_scope=Google Calender;read https://googleapis.calender.read
+     *
+     * @param context The authentication context with authentication request having the query parameters.
+     * @return  The scopes requested by the application via the query parameters for federated token sharing.
+     */
+    private String getRequestedScopesViaQueryParams(AuthenticationContext context) {
+
+        String authenticatorName = getFederatedAuthenticatorName(context);
+        if (StringUtils.isBlank(authenticatorName)) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("No external IDP name found in the authentication context for federated token sharing. " +
+                        "Cannot retrieve the query parameters.");
+            }
+            return null;
+        }
+
+        String scopeString = getQueryParameter(context, OIDCAuthenticatorConstants.FEDERATED_TOKEN_SCOPE);
+        if (StringUtils.isBlank(scopeString)) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("No query parameter " + OIDCAuthenticatorConstants.FEDERATED_TOKEN_SCOPE +
+                        " found in federated token sharing, IDP: " + authenticatorName);
+            }
+            return null;
+        }
+        /*
```

Comment:
lets add a debug log of the entire scopeString so we can identify any malformed things getting dropped.

Reply:
Fixed d2b40f9255b4a5ef3b8f9d44bd58ccdba9f50f70